lib/remote/httpserverconnection.hpp

```diff
@@ -31,15 +31,15 @@ class HttpServerConnection final : public Object
 	void Disconnect();
 	void StartStreaming();
 
-	bool Disconnected();
+	bool Disconnected() const;
 
 private:
 	ApiUser::Ptr m_ApiUser;
 	Shared<AsioTlsStream>::Ptr m_Stream;
 	double m_Seen;
 	String m_PeerAddress;
 	boost::asio::io_context::strand m_IoStrand;
-	bool m_ShuttingDown;
+	std::atomic<bool> m_ShuttingDown;
```

Comment:
```suggestion
	Atomic<bool> m_ShuttingDown;
```

Reply:
As @julianbrost pointed out https://github.com/Icinga/icinga2/pull/10214#issuecomment-2456968945, just using the default constructor of `std::atomic` is problematic. `Atomic` has no default constructor which enforces not (accidentally) using it (now and in future) at compile time (since #10215 🙈).

Reply:
This is C++. If we can easily enforce something at compile time, we do it.